src/main/java/de/tum/in/www1/artemis/domain/Attachment.java

```diff
@@ -83,7 +83,7 @@ public void onLoad() {
         if (attachmentType == AttachmentType.FILE && getLecture() != null && link != null && link.contains(Constants.FILEPATH_ID_PLACHEOLDER)) {
             link = link.replace(Constants.FILEPATH_ID_PLACHEOLDER, getLecture().getId().toString());
         }
-        if (attachmentType == AttachmentType.FILE && getAttachmentUnit() != null && link != null && link.contains(Constants.FILEPATH_ID_PLACHEOLDER)) {
+        else if (attachmentType == AttachmentType.FILE && getAttachmentUnit() != null && link != null && link.contains(Constants.FILEPATH_ID_PLACHEOLDER)) {
             link = link.replace(Constants.FILEPATH_ID_PLACHEOLDER, getAttachmentUnit().getId().toString());
```

Comment:
`Constants.FILEPATH_ID_PLACHEOLDER` has a typo. Maybe we can fix it with your PR?

Reply:
Done in 9a0a229d1accaf5134471f4873deb07e83db44df